src/client.ts

```diff
@@ -236,6 +238,18 @@ export interface ClientOptions {
    * Reference: https://gist.github.com/jed/982883
    */
   generateID?: () => ID;
+  /**
+   * An optional override for the JSON.parse function used to hydrate
+   * incoming messages to this client. Useful for parsing custom datatypes
+   * out of the incoming JSON.
+   */
+  reviver?: JSONReviver;
+  /**
+   * An optional override for the JSON.stringify function used to serialize
+   * outgoing messages from this client. Useful for serializing custom
+   * datatypes out to the client.
+   */
+  replacer?: JSONReplacer;
```

Comment:
I prefer longer, but more descriptive names. Especially for options which can grow:
```suggestion
  jsonMessageReviver?: JSONMessageReviver;
  /**
   * An optional override for the JSON.stringify function used to serialize
   * outgoing messages from this client. Useful for serializing custom
   * datatypes out to the client.
   */
  jsonMessageReplacer?: JSONMessageReplacer;
```